### DISCUSSION TITLE
How do I specify hostname in jobs.<job_id>.container.options

### DISCUSSION THREAD

**Discussion by jamiet-msm** [community/community#25760] [Actions]
<p>I am using a container to run some of my actions. Here is a portion of my workflow YAML:</p>
<pre><code class="lang-yaml">  sqlfluff-dataplatform:
    runs-on: [ self-hosted, ubuntu, mytag ]
    needs: [compute-pipeline-id, images-build]
    defaults:
      run:
        shell: bash
        working-directory: /mydir
    container:
      image: ${{needs.images-build.outputs.dataplatform-image-name-and-tag}}
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      - name: do stuff
        run: someCommand
</code></pre>
<p>It works fine. What I’d like to do is specify a hostname for the container. I’m pretty sure I can do this using <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idcontaineroptions" rel="noopener nofollow ugc">jobs.&lt;job_id&gt;.container.options</a> however I can’t figure out the syntax. I’ve tried this:</p>
<pre><code class="lang-yaml">  sqlfluff-dataplatform:
    runs-on: [ self-hosted, ubuntu, mytag ]
    needs: [compute-pipeline-id, images-build]
    defaults:
      run:
        shell: bash
        working-directory: /mydir
    container:
      image: ${{needs.images-build.outputs.dataplatform-image-name-and-tag}}
      options:
        hostname: somehostname
</code></pre>
<p>but that causes my workflow run to fail with error:</p>
<blockquote>
<p>[ <strong>Invalid workflow file</strong> : .github/workflows/ci.yml#L271]<br>
The workflow is not valid. .github/workflows/ci.yml (Line: 271, Col: 9): A mapping was not expected</p>
</blockquote>
<p>Can someone tell me what the correct synax is, because I cannot figure it out. Thx in advance.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>It should be a string like you’d use on the Docker command line, e.g.:</p>
<pre><code class="lang-yaml">container:
  image: some-image:latest
  options: --hostname somehostname
</code></pre>

**Comment by jamiet-msm** [comment]
<p>Oh, easy as that eh. Thanks.</p>

**Comment by jamiet-msm** [comment]
<p>I have a follow-up question here: <a href="https://github.community/t/how-do-i-refer-to-env-context-in-a-workflow-file/221328" class="inline-onebox">How do I refer to env context in a workflow file?</a> regarding use of env context in <code>job.&lt;job_id&gt;.container.options</code></p>
